### PR TITLE
[ticket/14073] Add several core events to the functions_admin.php

### PR DESCRIPTION
**Pull request by rxu**
Add core events to the functions move_posts(), delete_posts(), delete_attachments().

<a href="https://tracker.phpbb.com/browse/PHPBB3-14073">PHPBB3-14073</a>.

### CODE REVIEW

File: phpBB/includes/functions_admin.php [hunk 1]
```diff
@@ -610,6 +610,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
 		trigger_error('NO_TOPIC');
 	}
 
+	/**
+	 * Perform additional actions before the posts moving
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: before moving posts

File: phpBB/includes/functions_admin.php [hunk 2]
```diff
@@ -610,6 +610,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
...
+	 *
+	 * @event core.move_posts_before
+	 * @var	array	post_ids	Array of the moved post ids
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: Array of post ids to move

File: phpBB/includes/functions_admin.php [hunk 3]
```diff
@@ -610,6 +610,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
...
+	 * @var	string	topic_id	The topic id the posts are moved to
+	 * @var bool	auto_sync	Whether or not to perform auto sync
+	 * @var	array	forum_ids	Array of the forum ids the posts are moved from
+	 * @var	array	topic_ids	Array of the topic ids the posts are moved from
+	 * @var	array	forum_row	Array with the forum id of the topic the posts are moved to
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: Isn't that rather the forum data instead of the forum id?

File: phpBB/includes/functions_admin.php [hunk 4]
```diff
@@ -610,6 +610,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
...
+	 * @since 3.1.6-RC1
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: 3.1.7-RC1

File: phpBB/includes/functions_admin.php [hunk 5]
```diff
@@ -620,6 +642,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
 		WHERE " . $db->sql_in_set('post_msg_id', $post_ids);
 	$db->sql_query($sql);
 
+	/**
+	 * Perform additional actions after the posts moving
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: after moving posts

File: phpBB/includes/functions_admin.php [hunk 6]
```diff
@@ -620,6 +642,28 @@ function move_posts($post_ids, $topic_id, $auto_sync = true)
...
+	 *
+	 * @event core.move_posts_after
+	 * @var	array	post_ids	Array of the moved post ids
+	 * @var	string	topic_id	The topic id the posts are moved to
+	 * @var bool	auto_sync	Whether or not to perform auto sync
+	 * @var	array	forum_ids	Array of the forum ids the posts are moved from
+	 * @var	array	topic_ids	Array of the topic ids the posts are moved from
+	 * @var	array	forum_row	Array with the forum id of the topic the posts are moved to
+	 * @since 3.1.6-RC1
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: See above for forum_row and @since

File: phpBB/includes/functions_admin.php [hunk 7]
```diff
@@ -886,6 +930,32 @@ function delete_posts($where_type, $where_ids, $auto_sync = true, $posted_sync =
 
 	$table_ary = array(POSTS_TABLE, REPORTS_TABLE);
 
+	/**
+	* Perform additional actions during post(s) deletion before it's started
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: before running the queries

File: phpBB/includes/functions_admin.php [hunk 8]
```diff
@@ -886,6 +930,32 @@ function delete_posts($where_type, $where_ids, $auto_sync = true, $posted_sync =
...
+	*
+	* @event core.delete_posts_in_transaction_before
+	* @var	array	post_ids					Array with deleted posts' ids
+	* @var	array	poster_ids					Array with deleted posts' author ids
+	* @var	array	topic_ids					Array with deleted posts' topic ids
+	* @var	array	forum_ids					Array with deleted posts' forum ids
+	* @var	string	where_type					Variable containing posts deletion mode
+	* @var	mixed	where_ids					Array or comma separated list of posts ids to delete
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: post ids

File: phpBB/includes/functions_admin.php (hunk 3 above)
Comment (rxu) [inline phpBB/includes/functions_admin.php]: That array contains forum ids only.

**Comment by rxu** [comment]
@marc1706 Updated #3814.

**Pull request merged**